### PR TITLE
Adds download functionality

### PR DESCRIPTION
**Pull request by robschleusner**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -88,4 +119,117 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD).setAction(
...
+      } else if (linuxDistro === 'ubuntu20') {
+        return UBUNTU_20_CHECKSUM;
+      }
+      break;
+    case 'win32':
+      console.log('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', " Windows download is not supported yet, please compile SolidityX from source");
+      process.exit(1);
+      return WINDOWS_CHECKSUM;
+    default:
+      console.error('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', 'Unsupported OS');
```
Comment (Juuddi) [inline src/index.ts:210]: Might be worth it to display to the user the current supported operating systems if we're serving them an error that says their OS is unsupported.

File: src/index.ts [hunk 2]
```diff
@@ -88,4 +119,117 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD).setAction(
...
+    case 'linux':
+      const linuxDistro = await getLinuxDistribution();
+      if (linuxDistro === 'ubuntu18') {
+        return UBUNTU_18_CHECKSUM;
+      } else if (linuxDistro === 'ubuntu20') {
+        return UBUNTU_20_CHECKSUM;
+      }
+      break;
+    case 'win32':
+      console.log('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', " Windows download is not supported yet, please compile SolidityX from source");
```
Comment (Juuddi) [inline src/index.ts:206]: Including a link to the solidity compiler build from source instructions here: https://docs.soliditylang.org/en/latest/installing-solidity.html

Or maybe a link to the SolidityX repo build from source instructions as we continue to build them out for different OS: https://github.com/dominant-strategies/SolidityX?tab=readme-ov-file#the-solidity-contract-oriented-programming-language

Same goes for the rest of the instances where we tell the user `Your OS is not supported yet`

**Review by Juuddi — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -88,4 +119,117 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD).setAction(
...
+  console.log("3: Other")
+  console.log("Enter the number of your distribution:");
+
+  const choice = await askQuestion('> ');
+
+  switch (choice.trim().toLowerCase()) {
+    case '1':
+    case 'ubuntu18':
+    case 'ubuntu 18':
+      console.log('\x1b[1m\x1b[31m%s\x1b[0m%s', 'Error:', " Ubuntu 18 download is not supported yet, please compile SolidityX from source");
```
[the remark this answers is not in the document]
Comment (robschleusner) [reply src/index.ts:174]: Can't detect Ubuntu18 vs 20. Cna detect win vs mac vs ubuntu

File: src/index.ts [hunk 4]
```diff
@@ -88,4 +119,117 @@ internalTask(TASK_COMPILE_SOLIDITY_GET_SOLC_BUILD).setAction(
...
+      process.exit(1);
+      ubuntu_version = 'ubuntu18'
+      return 'ubuntu18';
+    case '2':
+    case 'ubuntu20':
+    case 'ubuntu 20':
```
[the remark this answers is not in the document]
Comment (robschleusner) [reply src/index.ts:180]: If they misspell then they would see a message "invalid selection" 

the options are ubuntu 18, ubuntu 20, and other. If they put in other then they would see sorry we don't support

File: dist/src/constants.d.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+declare const DARWIN_SOLX_LINK = "https://github.com/dominant-strategies/SolidityX/releases/download/0.8.19-solidityx/solc-osx";
+declare const DARWIN_CHECKSUM = "f667915f7b33ea463583e1d2bacea127e501b0413e5505806cc958506f1446e1";
```
[the remark this answers is not in the document]
Comment (robschleusner) [reply dist/src/constants.d.ts:2]: From what I've seen, darwin works for all MacOS versions. If it doesn't work for the user then they would have to compile it themselves

File: src/index.ts (hunk 2 above)
Comment (robschleusner) [reply src/index.ts:206]: Done

File: src/index.ts (hunk 1 above)
Comment (robschleusner) [reply src/index.ts]: Done

File: dist/src/constants.d.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Juuddi) [reply dist/src/constants.d.ts:2]: We'll just have to keep an eye on support requests after this release goes out to make sure that it doesn't break for different versions. This should be fine for now.

File: src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Juuddi) [reply src/index.ts:180]: Perfecto

**Review by Juuddi — APPROVED** [review]

**Pull request merged**